api/spec/v1.yaml

```diff
@@ -1579,6 +1581,11 @@ components:
             Hash of this event's originating transaction.
             Absent if the event did not originate from a transaction.
           example: *tx_hash_1
+        related_runtime:
```

Comment:
hm should we have a field for the id though? ugh

Reply:
I think it's fine not to. We _could_ remove `related_runtime` and `related_runtime_round` too. We provide the raw `body`, which includes the ID (and everything else). Note how e.g. we also don't have a field for `amount`, which is present in several common events.

Doing it this way puts a bigger onus on end users, including having to deal with versioning if node-emitted types change. But it makes it easier for us :). Given that the types are quite stable and that we don't have a lot of users, I think it's a fine tradeoff. We can always add more fields later if we want to.

Reply:
we got through maintaining these extra related_runtime and related_runtime_round columns. the latter one about round looks like it'll be pretty annoying to emulate on the api side. so let's keep these

Reply:
keeping this, providing the raw ID too

Reply:
thanks, discussing more there

Reply:
they're consistent with a naming scheme based on what they come from and consistent across the stack.

resolving this thread :smiling_face_with_tear:

Reply:
Decision from slack: Let's use just the `runtime_*` prefix (no `related_`) everywhere.

Reasoning, abridged:
> W: having a "related_..." field doesn't have to mean that it's a separate runtime though. if a transfer transaction has a "to" address and also a related_addresses list, we can have the same address in both
> and even if we had all four, they wouldn't be in conflict. hypothetically:
> - roothash runtime ID <- taken from the event field
> - roothash runtime name <- precomputed name from runtime ID field
> - related runtime name <- "computed," in practice by looking up the runtime from the runtime ID field
related runtime ID <- discretionary denormalization of related runtime name

> M: do we expect other sources of related runtimes? other than roothash I mean.

> W: certain staking events could mark a runtime as related, e.g. those from Deposit, Withdraw, (Un)Delegate

> M/W: (back and forth on how this might look and what info we'll need once we want to show Activities.)

> M: should we go with `related_*` everywhere? the raw `roothash_id` field is in the event body anyway

> W: it's not :/. Events look like
>    type = 'roothash.executor_commitment'
>    body = event.roothash.executor_commitment
>    runtime_id = event.roothash.runtime_id

> M: Dang. Back to Activities: Then how about we just go with roothash_* everywhere? My guess is that if/when we have runtime-triggered events (like in the case of Deposit and Delegate) and we want to group them into Activities, we'll want a way to _link_ those various events and txs and messages together, but not necessarily to propagate properties (like runtime) from one event/tx to others.

> W/M: SGTM, good enough guess. Also, it gives us freedom to expand to an additional `related_*` later if needed.

Reply:
renamed. kind of unpleasant now that it's unclear from the name how they're different. but it's more of nexus's style not to distinguish anyway